src/core.c/Lock.pm6

```diff
@@ -9,7 +9,9 @@ my class Lock {
         method new() {
             X::Lock::ConditionVariable::New.new.throw
         }
-        method wait() { nqp::condwait(self) }
+        proto method wait(|) {*}
+        multi method wait() { nqp::condwait(self) }
+        multi method wait(&predicate) { nqp::condwait(self) until predicate; }
```

Comment:
Hm, I doubt these return anything useful or that we've spec'd that, but I'd suggest while we're at this, we throw a `--> Nil` on the two candidates.

Reply:
Done